arklib/src/lib.rs

```diff
@@ -558,7 +558,7 @@ pub mod android {
         trace!("providing index for root {}", &root_string);
 
         let root: PathBuf = PathBuf::from(root_string);
-        arklib::provide_index(&root)
+         Ok(arklib::provide_index(&root)?)
```

Comment:
Fix mismatched types build error.

![Screenshot 2023-10-31 at 21 48 33](https://github.com/ARK-Builders/arklib-android/assets/4281969/2db5d319-fb8a-4481-91ee-273c728fe5f9)

Reply:
That's interesting.. Must affect all systems, not only macOS

Reply:
Are you sure it's not related to the commit change of arklib version ? 
@tuancoltech it works with cargo-ndknand the previous commit??

Reply:
`Are you sure it's not related to the commit change of arklib version ?`
-> @gwendalF For this specific error, I'm not sure if it's related to the change of arklib version, as we could not even be able to reach this point without the change.
` it works with cargo-ndknand the previous commit??`
-> With `cargo-ndk`, both me and @Rizary do not encounter this error.

Reply:
@tuancoltech So 
`arklib::provide_index(&root)`
and
 `Ok(arklib::provide_index(&root)?)`
works with cargo ndk?
For me it's related to the version change of the arklib version which change the signature of the provide_index function. But I don't know why there is no issue with cargo ndk

Reply:
@gwendalF : Yes.

If you want to try yourself, here're the steps I took to build the lib via `cargo-ndk`:

```
- cd arklib
- cargo ndk -o ./jniLibs build
- cd ..
- ./gradlew lib:assemble
```

This successfully builds the `aar` in my side in both branches.